app/javascript/mixins/tourRoot.js

```diff
@@ -0,0 +1,77 @@
+import TourHolder from "@components/tour/TourHolder.vue"
+import axios from 'axios'
+
+/**
+ * Adds the data and methods necessary for being the parent component to tour components {@link module:components/tour/TourHolder TourHolder}
+ * @mixin tourRoot
+ * @vue-data {Object[]} tourSteps=null the steps of the tour
+ * @vue-data {Boolean} hasTour=false does the page have a tour?
+ * @vue-data {Object} tourData=null the data returned from a request for the page's tour
+ * @vue-computed {String} tourPage the string representation of the current page's path for requesting and closing the tour
+ * @listens tour-stopped
+ */
+export default {
+	data: function() {
+		return {
+			tourSteps: null,
+			hasTour: false,
+			tourData: null
+		}
+	},
+	computed: {
+		tourPage() {
+			return `/tours/${window.location.pathname.replace(/\//g, "-")}.json`
```

Comment:
so, for `domain.com/foo/bar` this would return `/tours/-foo-bar.json` unless im reading that wrong.
why the leading dash?

it might be cleaner to split that up into a proper function instead of a one-liner

Reply:
the leading dash allows me to also have a "name" for the root. I'll make this two lines, though